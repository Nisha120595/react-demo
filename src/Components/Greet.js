import React from 'react';

// function Greet(){
//     return <h1>Hello Nisha Rajput</h1>
// }


// export const Greet = props => {
// return (
//      <div>
//           <h1>Hello {props.name} a.k.a {props.heroName}</h1>
//           {props.children}
//     </div>
//   )
// }


// export const Greet = ({name,heroName}) => { // Destructuring first way
//   return (
//        <div>
//             <h1>Hello {name} a.k.a {heroName}</h1>
//       </div>
//     )
// }

export const Greet = props => { // Destructuring second way
  const {name,heroName} = props;
  return  <h1 data-testid="greetings" className='greetings-class'>Hi {name} a.k.a {heroName}</h1>
}

//export default Greet
