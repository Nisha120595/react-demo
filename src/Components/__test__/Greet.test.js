import React from 'react';
import ReactDOM from 'react-dom';
import { Greet } from './../Greet';
import { render , cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import renderer from 'react-test-renderer';

afterEach(cleanup);

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Greet />,div);
})

it('renders file properly', () => {
    const { getByTestId } = render(<Greet name = "Bruce" heroName = "Batman"/>);
    expect(getByTestId('greetings')).toHaveTextContent("Hi Bruce a.k.a Batman");
})

it('renders file properly', () => {
    const { getByTestId } = render(<Greet name = "Diana" heroName = "Wonder woman"/>);
    expect(getByTestId('greetings')).toHaveTextContent("Hi Diana a.k.a Wonder woman");
})

it('matches snapshot', () => {
    const tree = renderer.create(<Greet name = "Diana" heroName = "Wonder woman"/>).toJSON();
    expect(tree).toMatchSnapshot();
})

it('matches snapshot', () => {
    const tree = renderer.create(<Greet name = "Bruce" heroName = "Batman"/>).toJSON();
    expect(tree).toMatchSnapshot();
})