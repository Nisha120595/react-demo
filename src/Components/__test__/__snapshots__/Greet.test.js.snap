// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<h1
  className="greetings-class"
  data-testid="greetings"
>
  Hi 
  Diana
   a.k.a 
  Wonder woman
</h1>
`;

exports[`matches snapshot 2`] = `
<h1
  className="greetings-class"
  data-testid="greetings"
>
  Hi 
  Bruce
   a.k.a 
  Batman
</h1>
`;
